src/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanner.java

```diff
@@ -109,22 +125,46 @@ public int getPluginId() {
 		return PLUGIN_ID;
 	}
 	
-	private String getDescription(boolean isMissing) {
-		if (isMissing) //Not set at all
-			return Constant.messages.getString(MESSAGE_PREFIX + "missing.desc");
-		else //Set improperly?
-			return Constant.messages.getString(MESSAGE_PREFIX + "desc");
-	}
-
-	private String getOtherInfo() {
-		return Constant.messages.getString(MESSAGE_PREFIX + "otherinfo");
+	private String getAlertElement(VulnType currentVT, String element) {
+		String elementValue="";
+		switch (currentVT) {
+			case XFO_MISSING:
+				elementValue=Constant.messages.getString(MESSAGE_PREFIX + "missing." + element);

```

Comment:
The case statements could return the value returned by getString instead of using an extra variable.
A default case (that returns an empty string) could be added, making elementValue unnecessary.